Prefer Python3 for the remotely-executed Python script

heapcrash opened the pull request:
This should be done as part of our general transition to Python3,
but should also be done because some systems with e.g. Python2.6
do not support OrderedDict, but DO have a Python3 version that works.

heapcrash committed:
Prefer Python3 for the remotely-executed Python script

This should be done as part of our general transition to Python3,
but should also be done because some systems with e.g. Python2.6
do not support OrderedDict, but DO have a Python3 version that works.

Arusekk commented on pwnlib/tubes/ssh.py:
```diff
@@ -932,6 +932,14 @@ def func(): pass
+        # This breaks things if a Python3 interpreter is selected.
+        executable = six.ensure_str(executable)
+        argv = [bytes(a) for a in argv]
+        env = None if not env else { six.ensure_str(k): six.ensure_str(v) for k,v in env}
```
Nah, the whole point of using bytearray was to preserve the bits, since bytearray always has a representation of `bytearray(b'whatever')`. This was used in order to support just everything that has a bytearray (python 2.6 up).

Arusekk commented on pwnlib/tubes/ssh.py:
```diff
@@ -942,7 +950,7 @@ def func(): pass
     integer_types = int,
 exe   = %(executable)r
-argv  = [bytes(a) for a in %(argv)r]
+argv  = %(argv)r
```
I wrote that here since the os.exec* functions complain about bytearray not being bytes. So it was the safest I guess.

If we really want to support everything everything, then we would have to convert all the things to some hex sequences or using pickle. And to deserialize it in some super-python1-compliant way in the best case and the bestestest would be to support deserializing and launching the target executable using perl, awk, and other weird utilities. But I really think that anything beyond py2.6 is at least as dead as [QMAGIC](https://github.com/torvalds/linux/blob/master/include/uapi/linux/a.out.h).

Arusekk reviewed (changes requested):
I believe that prepending python3 would be just enough.

Arusekk committed:
Merge branch 'dev' into prefer-python3-for-ssh-remote-script

Arusekk committed:
Make the script actually work in py2 and py3

Arusekk committed:
Fix & update CHANGELOG.md

This actually is a behavior change, and can be a breaking change for some,
so must be stated prominently in the changelog.
